src/playground-project.ts

```diff
@@ -223,79 +263,48 @@ export class PlaygroundProject extends LitElement {
     this._compileProject();
   }
 
-  private async _startWorkers() {
-    await Promise.all([
-      this._startTypeScriptWorker(),
-      this._installServiceWorker(),
-    ]);
+  firstUpdated() {
+    const worker = new Worker(typescriptWorkerScriptUrl);
+    this._typescriptWorkerAPI = wrap<TypeScriptWorkerAPI>(worker);
   }
 
-  private async _startTypeScriptWorker() {
-    if (this._typescriptWorkerAPI === undefined) {
-      const worker = new Worker(typescriptWorkerScriptUrl);
-      this._typescriptWorkerAPI = wrap<TypeScriptWorkerAPI>(worker);
-    } else {
-      console.debug('typescript-worker already started');
-    }
-  }
-
-  private async _installServiceWorker() {
-    if (!('serviceWorker' in navigator)) {
-      // TODO: show this in the UI
-      console.warn('ServiceWorker support required for <playground-project>');
+  private _onServiceWorkerProxyIframeLoad() {
+    const window = this._iframe.contentWindow;
+    if (!window) {
+      console.error('IFrame did not have window');
```

Comment:
Use `console.assert()` or throw? Can the message be a bit more descriptive? which iframe in particular.

Reply:
Done. I don't think this error will ever actually happen, but can't be sure.